config/puma/production.rb

```diff
@@ -27,6 +27,22 @@
 #
 preload_app!
 
+# Check if FOREMAN_BIND was set to an IP address based on the previous
+# definition of FOREMAN_BIND. If it is an IP address, define the host
+# and port through Puma's DSL. Otherwise rescue and assume the new
+# style of a fully specified bind in the formats of:
+#
+#  * unix:///run/foreman.sock
+#  * tcp://127.0.0.1:3000
+#
+begin
+  IPAddr.new(ENV['FOREMAN_BIND'])
+
+  port ENV.fetch('FOREMAN_PORT', '3000'), ENV['FOREMAN_BIND']
```

Comment:
I haven't tested this. Does `port` accept a string or does it need to be an integer? Probably doesn't hurt to use `to_i`.
```suggestion
  port ENV.fetch('FOREMAN_PORT', '3000').to_i, ENV['FOREMAN_BIND']
```

Reply:
I don't think it matters -- https://github.com/puma/puma/blob/4.3.6/lib/puma/dsl.rb#L193

Reply:
FWIW, I would've written `bind "tcp://HOST:PORT"` instead, as I find the `port` syntax confusing. But both obviously work the same in the end.